src/Taylor1.jl

```diff
@@ -666,3 +669,53 @@ function deriv{T<:Number}(a::Taylor1{T}, n::Int=1)
     @assert a.order >= n >= 0
     factorial( widen(n) ) * a.coeffs[n+1] :: T
 end
+
+
+"""
+    A_mul_B!(y, a, b)
+
+Multiply a*b and save the result in y.  Extends y to fit the results if necessary.
+"""
+function A_mul_B!{T<:Number}(y::Vector{Taylor1{T}},a::Matrix{T},b::Vector{Taylor1{T}})
+
+    # determine the maximal order of b
+    order = maximum([b1.order for b1 in b])
+
+    n,k = size(a)
+
+    # extend y to fit all the coefficients and zero the
```

Comment:
Ok, I'll do the changes. It would be nice if you can also benchmark the changes.

As for your comment on `similar`, the following in master (inspired in your PR, but without it!) currently works; note that I am *not* creating `Y1`, and `Y` is obtained from the operation:
```julia
julia> using TaylorSeries

julia> order = 30;

julia> n1 = 4;

julia> k1 = 3;

julia> B1 = randn(n1,order);

julia> A1  = randn(k1,n1);

julia> B  = Taylor1{Float64}[Taylor1(collect(B1[i,1:i]),i) for i=1:n1];

julia> Y = A_mul_B!(similar(B, k1), A1, B)
3-element Array{TaylorSeries.Taylor1{Float64},1}:
  0.27316308514920934 + 2.3361726531529268 t + 1.2648372715206755 t² + 0.04987013186300208 t³ + 𝒪(t⁵)
      1.896823400442119 - 0.8006875447982573 t - 2.232382532282572 t² + 0.8926690617877661 t³ + 𝒪(t⁵)
   - 0.2830616479144453 - 3.470193293491136 t - 5.1783732101128725 t² + 2.8494688946414692 t³ + 𝒪(t⁵)

```

The result is at most of order 5 because that is the maximum order of `B`.